app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -153,9 +159,44 @@ class BrowserToolbarIntegration(
         toolbar.display.setUrlBackground(urlBackground)
     }
 
+    private fun setActionButtons(context: Context) {
+        val eraseAction = BrowserToolbar.Button(
+            imageDrawable = AppCompatResources.getDrawable(
+                context,
+                R.drawable.mozac_ic_delete
+            )!!,
+            contentDescription = context.getString(R.string.content_description_erase),
+            iconTintColorResource = R.color.primaryText,
+            listener = {
+                val openedTabs = store.state.tabs.size
+                TabCount.eraseButtonTapped.record(TabCount.EraseButtonTappedExtra(openedTabs))
+                eraseActionListener.invoke()
+            }
+        )
+        toolbar.addNavigationAction(eraseAction)
+
+        tabsCounterScope = store.flowScoped { flow ->
+            flow.ifChanged { state -> state.tabs.size > 1 }
+                .collect { state ->
+                    val tabsAction = TabCounterToolbarButton(
+                        lifecycleOwner = fragment,
+                        showTabs = {
+                            toolbar.hideKeyboard()
+                            tabCounterListener.invoke()
+                        },
+                        store = store
+                    )
+                    if (state.tabs.size > 1) {
+                        toolbar.addBrowserAction(tabsAction)
+                    } else {
+                        toolbar.removeBrowserAction(tabsAction)
+                    }
+                }
+        }
```

Comment:
There's a mismatch now. We subscribe in `init()`, but we unsubscribe in `stop()`. The start and stop methods are getting called with the lifecycle. So if you open a tab, minimize the app, resume the app and then open another tab, then we will have unsubscribed and will not add the button.

We could observe/cancel from start/stop, like we do for the security indicator changes. But then we also need to make sure we do not add the button multiple times. :)

Reply:
I tried to reproduce this scenario but I didn't succeed. Do you think that you can help me with a video?

Reply:
https://user-images.githubusercontent.com/89638/141994422-310af25b-595c-43d1-8365-a7c21644405e.mp4